docs/reference/extensions/expressjs-framework-extension.rst

```diff
@@ -0,0 +1,154 @@
+.. _expressjs-framework-extension:
+
+
+ExpressJS framework extension
+=============================
+
+The ``expressjs-framework`` extension includes configuration options customised
+for a Express application. This document describes all the keys that a user
+may interact with.
+
+.. tip::
+
+    If you'd like to see the full contents contributed by this extension,
+    see :ref:`How to manage extensions <manage-extensions>`.
+
+
+``charmcraft.yaml`` > ``config`` > ``options``
+----------------------------------------------
+
+You can use the predefined options (run ``charmcraft expand-extensions`` for details)
+but also add your own, as needed.
+
+The predefined configuration options for the ``expressjs-framework`` are:
+
+* **port**: Port in which the application should listen. The ingress will be configured
+  using this port. The environment variable passed to the app is ``PORT``. Default value
+  is 8080.
+
+* **app-secret-key**: A secret you can use for sessions, csrf or any other thing
+  where you need a random secret shared by all units. The environment variable passed to
+  the app is ``APP_SECRET_KEY``. The default value is a 64 byte Base64 encoded random
+  string.
```

Comment:
It'd be much better if we pushed `app-secret-key-id` as the correct way to do this. Either that one alone, or at least both of the options with clear guidance that it's best to use a Juju secret.